tensorboard/webapp/settings/_redux/settings_effects_test.ts

```diff
@@ -0,0 +1,132 @@
+        SettingsEffects,
+      ],
+    }).compileComponents();
+
+    store = TestBed.inject<Store>(Store) as MockStore;
+    store.overrideSelector(getReloadEnabled, false);
+    store.overrideSelector(getReloadPeriodInMs, 30000);
+    store.overrideSelector(getPageSize, 5);
+    spyOn(store, 'dispatch').and.callFake((action: Action) => {
+      dispatchedActions.push(action);
+    });
+    const effects = TestBed.inject(SettingsEffects);
+    effects.initialSavedSettingsFetch$.subscribe();
+    effects.saveReloadEnabled$.subscribe();
+    effects.saveReloadPeriodInMs$.subscribe();
+    effects.savePageSize$.subscribe();
+    dataSource = TestBed.inject(SettingsDataSource);
+  });
+
+  it('fetches saved settings on init and dispatches returned result', () => {
+    const fetchSavedSettings = spyOn(dataSource, 'fetchSavedSettings');
+    const subject = new Subject<Partial<Settings>>();
+    fetchSavedSettings.and.returnValue(subject);
+
+    actions.next(TEST_ONLY.initAction());
+    // While waiting for the fetchSavedSettings function to return, a single
+    // action has been emitted.
+    expect(dispatchedActions).toEqual([fetchSavedSettingsRequested()]);
+    // Simulate fetchSavedSettings returning.
+    subject.next({reloadEnabled: true});
+    // An additional action has been emitted.
+    expect(dispatchedActions).toEqual([
+      fetchSavedSettingsRequested(),
+      fetchSavedSettingsSucceeded({savedSettings: {reloadEnabled: true}}),
+    ]);
+  });
+
+  it('dispatches failure when fetch fails', () => {
+    const fetchSavedSettings = spyOn(dataSource, 'fetchSavedSettings');
+    fetchSavedSettings.and.returnValue(throwError(new Error('Request failed')));
+
+    actions.next(TEST_ONLY.initAction());
+    expect(dispatchedActions).toEqual([
+      fetchSavedSettingsRequested(),
+      fetchSavedSettingsFailed(),
+    ]);
+  });
+
+  it('saves reloadEnabled', () => {
+    const saveReloadEnabled = spyOn(dataSource, 'saveReloadEnabled');
+
+    store.overrideSelector(getReloadEnabled, false);
+    store.refreshState();
+    actions.next(toggleReloadEnabled());
+    expect(saveReloadEnabled).toHaveBeenCalledWith(false);
+
+    store.overrideSelector(getReloadEnabled, true);
+    store.refreshState();
+    actions.next(toggleReloadEnabled());
+    expect(saveReloadEnabled).toHaveBeenCalledWith(true);
+  });
+
+  it('saves reloadPeriodInMs', () => {
+    const saveReloadPeriodInMs = spyOn(dataSource, 'saveReloadPeriodInMs');
+
+    store.overrideSelector(getReloadPeriodInMs, 1111);
+    store.refreshState();
+    actions.next(changeReloadPeriod({periodInMs: 2222}));
+    expect(saveReloadPeriodInMs).toHaveBeenCalledWith(1111);
+  });
+
+  it('saves reloadPeriodInMs', () => {
+    const savePageSize = spyOn(dataSource, 'savePageSize');
+
+    store.overrideSelector(getPageSize, 1111);
+    store.refreshState();
+    actions.next(changePageSize({size: 2222}));
+    expect(savePageSize).toHaveBeenCalledWith(1111);
```

Comment:
Not sure if I understand this. Why `savePageSize` method is called with `1111` not `2222`?

Reply:
Good question. Added comments. Let me know if it needs more clarity.